lsp/process_order.js

```diff
@@ -0,0 +1,423 @@
+        },
+        cbk);
+      }],
+
+      getChainHeight: ['getMessage', ({getMessage}, cbk) => {
+        if (!getMessage.message) {
+          return cbk();
+        }
+
+        return getHeight({lnd: args.lnd}, cbk);
+      }],
+
+      // Get chain fees
+      getChainFees: ['getMessage', ({getMessage}, cbk) => {
+        // Exit early when there is no message
+        if (!getMessage.message) {
+          return cbk();
+        }
+
+        const blocks = getMessage.message.params.confirms_within_blocks;
+
+        return getChainFeeRate({confirmation_target: Number(blocks), lnd: args.lnd}, cbk);
+      }],
+
+      // Calculate fees
+      getFees: ['getChainFees', 'getMessage', ({getChainFees, getMessage}, cbk) => {
+        // Exit early when there is no message
+        if (!getMessage.message) {
+          return cbk();
+        }
+
+        const rate = getChainFees.tokens_per_vbyte;
+        const {message} = getMessage;
+
+        // Aprox for 2 inputs and 2 outputs
+        const vbytes = 300;
+
+        // Fees per quarter
+        const fees = (args.fee_rate / 1e6) * Number(message.params.lsp_balance_sat);
+
+        // Total fees with opening and closing costs
+        const total = fees + (rate * vbytes * 2);
+
+        return cbk(null, {fees: total});
+      }],
+
+      // Send order message
+      makeOrder: ['getFees', 'getMessage', async ({getFees, getMessage}) => {
+        // Exit early when there is no message
+        if (!getMessage.message) {
+          return;
+        }
+
+        try {
+          const orderResponse = responses.lsps1CreateOrderResponse;
+          const {message} = getMessage;
+          const {fees} = getFees;
+          const orderId = makeOrderId();
+          const expiresAt = expiryDate(orderExpiryMs);
+          orderResponse.result.order_id = orderId;
+          orderResponse.result.lsp_balance_sat = message.params.lsp_balance_sat;
+          orderResponse.result.client_balance_sat = constants.maxPushAmount;
+          orderResponse.result.confirms_within_blocks = message.params.confirms_within_blocks;
+          orderResponse.result.channel_expiry_blocks = constants.channelExpiryBlocks;
+          orderResponse.result.created_at = currentDate();
+          orderResponse.result.expires_at = expiresAt;
+          orderResponse.result.announce_channel = message.params.announce_channel;
+          orderResponse.result.order_state = constants.orderStates.created;
+          orderResponse.result.payment.state = constants.paymentStates.expectPayment;
+          orderResponse.result.payment.fee_total_sat = String(fees);
+  
+          // Same as fees because push amounts are not supported
+          orderResponse.result.payment.order_total_sat = String(fees);
+
+          const secret = randomSecret();
+          const invoiceId = sha256(secret);
+  
+          const {request, id} = await createHodlInvoice({expires_at: expiresAt, id: invoiceId, lnd: args.lnd, tokens: fees});
```

Comment:
you don't need to make a secret and invoiceId, the secret can be generated by the method itself